### PR TITLE
Add polygon support

### PR DESCRIPTION
**Pull request by alchemistgo87**

### CODE REVIEW

File: src/app/(project)/projects/[project]/drops/[drop]/holders/Holders.tsx [hunk 1]
```diff
@@ -125,17 +125,31 @@ export default function Holders({ project, drop, loading }: HoldersProps) {
...
+                        className="flex gap-2 items-center"
+                      >
+                        <Icon.ExternalLink /> <span>View on SolScan</span>
+                      </Link>
+                    );
+                    break;
+                  case Blockchain.Polygon:
+                    options.push(
+                      <Link
+                        href={`https://polygonscan.com/address/${address}`}
```
Comment (kespinola) [inline src/app/(project)/projects/[project]/drops/[drop]/holders/Holders.tsx:144]: The address is `{contractAddress}:{tokenId}` so we need to replace the `:` with a `/` and the url is 

https://polygonscan.com/nft/{contractAddress}/{tokenId}

This is a good candidate for a client field on the object of exploreLink that handles figuring encapsulates the link assembly logic.

And then have the label be `View on explorer` for all of them.

File: src/app/(project)/projects/[project]/drops/[drop]/mints/Mints.tsx [hunk 2]
```diff
@@ -165,18 +165,34 @@ export default function Mints({ loading, project, drop }: MintsProps) {
...
-                    <Link
-                      href={`https://solscan.io/tx/${txId}`}
-                      target="_blank"
-                      key="change_email"
-                      className="flex gap-2 items-center"
-                    >
-                      <Icon.ExternalLink /> <span>View on SolScan</span>
-                    </Link>
-                  );
+                if (txId) {
```
Comment (kespinola) [inline src/app/(project)/projects/[project]/drops/[drop]/mints/Mints.tsx:168]: Same with having client field of `transactionLink`.

File: src/app/(drops)/projects/[project]/drops/new/details/page.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ const BLOCKCHAIN_LABELS = {
   [Blockchain.Polygon]: 'Polygon',
 };
 
-const BLOCKCHAIN_OPTIONS = [Blockchain.Solana];
+const BLOCKCHAIN_OPTIONS = [Blockchain.Solana, Blockchain.Polygon];
```
Comment (kespinola) [inline src/app/(drops)/projects/[project]/drops/new/details/page.tsx:22]: We need some extra validation logic for polygon. The differences are:

Can only have 1 creator.
Can remove the max length constraints on name and symbol so switch the maxLength validation with a custom one.

And if you return a string it should show the error banner to resolve this bug ticket at the same time.

https://github.com/holaplex/hub/issues/215

**Review by kespinola — CHANGES_REQUESTED** [review]
Can you double check the wallet views as well for the customers and treasuries. I think they need to be updated.

File: src/app/(project)/projects/[project]/drops/[drop]/holders/Holders.tsx [hunk 4]
```diff
@@ -124,19 +138,19 @@ export default function Holders({ project, drop, loading }: HoldersProps) {
...
-                      <Icon.ExternalLink /> <span>View on SolScan</span>
-                    </Link>
-                  );
-                }
+                const exploreLink = new ExploreLink(
+                  holdersQuery.data?.project.drop?.collection.blockchain as Blockchain
+                );
+                options.push(
+                  <Link
+                    href={exploreLink.getAccountLink(address) as string}
```
[the remark this answers is not in the document]
Comment (alchemistgo87) [reply src/app/(project)/projects/[project]/drops/[drop]/holders/Holders.tsx:146]: Oh, Ok.

[the remark this answers is not in the document]
Comment (alchemistgo87) [reply src/app/(project)/projects/[project]/drops/[drop]/holders/Holders.tsx:146]: The blockchain field is in the collection data which is parent object of holders. I found no option to get access to the parent data inside the apollo client.

[the remark this answers is not in the document]
Comment (kespinola) [reply src/app/(project)/projects/[project]/drops/[drop]/holders/Holders.tsx:146]: Looks like you get the cache as an argument so can get the collection mint from the holder and the the collection from the collection mint.

Its a little unintuitive so I'll refactor it and you can have as an example. This is good fro now. Ty.

**Pull request merged**